### PR TITLE
Improve API documentation in angular.dev for new signal APIs

### PR DESCRIPTION
**Pull request by devversion**
See individual commits. This is a start to help represent these APIs better in the angular.dev site without fundamentally changing how APIs are displayed in angular.dev (this would be a larger change)

image

Related prerequisite PR that is already merged: https://github.com/angular/dev-infra/pull/1854

### CODE REVIEW

**Comment by JeanMeche** [comment]
Screenshot 2024-03-19 at 01 50 39
Should we maybe make the items `white-space: nowrap;` ?

**Review by JeanMeche — APPROVED** [review]
Thx for this change ! 
One small nit that could be adressed separately

**Review by crisbeto — APPROVED** [review]

File: packages/core/src/authoring/queries.ts [hunk 1]
```diff
@@ -187,6 +191,9 @@ export interface ContentChildFunction {
  *   headerRequired = contentChild.required(MyHeader);            // Signal<MyHeader>
  * }
  * ```
+ *
+ * @initializerApiFunction
```
[the remark this answers is not in the document]
Comment (devversion) [reply packages/core/src/authoring/queries.ts:195]: I think the separate tag is probably better, as `@type` suggests an actual runtime type for IDEs, or in JSDoc— and additionally a separate tag allows us to accept additional metadata

**Review by sonukapoor — APPROVED** [review]
Nice 👍

**Comment by devversion** [comment]
Note: The changes to the generics of `input` are just naming. We can iterate more on these in the future. I've been discussing a couple of alternatives with Alex, but those are breaking.

**Review by AndrewKushnir — APPROVED** [review]
Reviewed-for: public-api

**Review by pkozlowski-opensource — APPROVED** [review]
LGTM

Reviewed-for: public-api

**Comment by dylhunn** [comment]
This PR was merged into the repository by commit e53e36bba9199bc2c68e774e05fb1b3d67b84283.

**Pull request closed without merging**